backend/app/views/spree/admin/option_types/_option_value_fields.html.erb

```diff
@@ -7,5 +7,12 @@
   </td>
   <td class="name"><%= f.text_field :name %></td>
   <td class="presentation"><%= f.text_field :presentation %></td>
-  <td class="actions"><%= link_to_remove_fields t('spree.actions.remove'), f, no_text: true %></td>
+  <td class="actions">
+    <% if f.object.persisted? %>
+      <%= link_to_delete f.object, no_text: true %>
+    <% else %>
+      <%= link_to_with_icon('trash', 'remove', '#', no_text: true, class: "spree_remove_fields",
+            data: { action: 'remove' }, title: t('spree.actions.remove')) %>
+    <% end %>
```

Comment:
How about extracting this into a helper, so that it can be easily reused?

Reply:
I like that, but i am unsure why 
`backend/app/views/spree/admin/option_types/_option_value_fields.html.erb`  
does not use the can? helpers like 
`backend/app/views/spree/admin/product_properties/_product_property_fields.html.erb`

then it should be that functionality to be extracted:
```
    <% if f.object.persisted? %>
      <% if can?(:destroy, f.object) %>
        <%= link_to_delete f.object, no_text: true %>
      <% end %>
    <% else %>
      <%= link_to_with_icon('trash', 'remove', '#', no_text: true, class: "spree_remove_fields",
            data: { action: 'remove' }, title: t('spree.actions.remove')) %>
    <% end %>
```
Also we should use then
`can?(:update_positions, f.object)`
there in `backend/app/views/spree/admin/option_types/_option_value_fields.html.erb` as well.

The PR is about Unifying the Handling of Option Values and Product Properties List.
This i would miss..
Or are there any caveats with using the can helpers there?

Reply:
Hmm, yeah, I hadn't noticed one uses `can?` while the other doesn't. [Admins can manage all models](https://github.com/solidusio/solidus/blob/master/core/lib/spree/permission_sets/super_user.rb), so I think we can safely add the `can?` statement here. Can you try and see if anything breaks?

As for the `update_positions` action, I think we could also add the `can?` for that one, although I'm not sure that change is in the scope of this PR.

Reply:
I'm in favor of extracting, but I'd use a partial instead.

Reply:
Thanks @aldesantis @kennyadsl. i am quite unsure. Please let me know what to do, you have a deeper understanding of the Rules and Code Structure of the Backend:

1. Leave it as is for this PR
2. add and test can? for option_value_fields and leave it for this PR
3. add and test can? for option_value_fields and put it all in a partial (in shared folder)
4. Just put the "uncanned" version in a partial.

Aside from that: Putting it in a partial could be an additional commit, right?
i would vote for 1 and would be happy to tackle additional challenges (like remove dragging from nonpersistent elements and put funcionality in partials) in following PRs.

Reply:
I'm also ok for merging this as is and iterate with new PRs when we see something that we can improve more.

Reply:
@aldesantis is this ok for you as well?

Reply:
Sounds good to me.